pkg/virt-api/webhooks/validating-webhook/admitters/validate-k8s-utils.go

```diff
@@ -53,6 +56,29 @@ var nodeFieldSelectorValidators = map[string]func(string, bool) []string{
 	metav1.ObjectNameField: ValidateNodeName,
 }
 
+// validateHostNameNotConformingToDNSLabelRules is function that validate spec.hostName and checks if given affinities are valid
+func validateHostNameNotConformingToDNSLabelRules(field *field.Path, hostName string) (causes []metav1.StatusCause) {
```

Comment:
In order to apply "utils" I'd suggest to make this function more generic.
```suggestion
func validateFieldValueNotConformingToDNSLabelRules(field *field.Path, value string) (causes []metav1.StatusCause) {
```

And of course to adjust `appendNewStatusCauseForHostNameNotConformingToDNSLabelRules` as well.
Also, I'd suggest to adjust the function to receive the exact field instead of the parent; in this way the
`appendNewStatusCauseForHostNameNotConformingToDNSLabelRules` function does not need to to `field.Child("hostname").String()` but `field.String()` is enough.